File: backend/app/schemas/drawing.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel


class DrawingCreate(BaseModel):
    title: str


class DrawingRename(BaseModel):
    title: str


class DrawingSceneUpdate(BaseModel):
    scene: dict[str, Any]


class Drawing(BaseModel):
    id: str
    note_id: str | None = None
    title: str
    created_at: datetime
    updated_at: datetime


class DrawingScene(Drawing):
    scene: dict[str, Any]

File: backend/src/types/drawing.ts
export interface DrawingCreate {
  title: string;
}

export interface DrawingRename {
  title: string;
}

export interface DrawingSceneUpdate {
  scene: Record<string, any>;
}

export interface Drawing {
  id: string;
  note_id?: string | null;
  note_title?: string | null;
  title: string;
  created_at: string;
  updated_at: string;
}

export interface DrawingScene extends Drawing {
  scene: Record<string, any>;
}
